### PR TITLE
Expand BuildOperationFinishedNotification interface with a details object.

### PR DESCRIPTION
**Pull request by felipefzdz**
Small improvement to https://github.com/gradle/gradle/pull/1961

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/internal/operations/BuildOperationNotificationBridge.java [hunk 1]
```diff
@@ -176,8 +178,17 @@ public Throwable getNotificationOperationFailure() {
...
+
+        @Override
         public String toString() {
-            return "BuildOperationFinishedNotification{id=" + id + ", result=" + result + '}';
+            return "Finished{"
+                + "id=" + id
+                + ", result=" + result
+                + ", details=" + details
+                + '}';
         }
```
Comment (etiennestuder) [inline subprojects/core/src/main/java/org/gradle/internal/operations/BuildOperationNotificationBridge.java]: `failure` is missing

File: subprojects/core/src/main/java/org/gradle/internal/operations/BuildOperationNotificationBridge.java [hunk 2]
```diff
@@ -153,11 +153,13 @@ public String toString() {
...
         private final Throwable failure;
+        private final Object details;
 
-        private Finished(Object id, Object result, Throwable failure) {
+        private Finished(Object id, Object result, Throwable failure, Object details) {
             this.id = id;
             this.result = result;
             this.failure = failure;
+            this.details = details;
         }
```
Comment (etiennestuder) [inline subprojects/core/src/main/java/org/gradle/internal/operations/BuildOperationNotificationBridge.java]: Switch order to have `details` after the `id`.

File: subprojects/core/src/main/java/org/gradle/internal/operations/BuildOperationNotificationBridge.java [hunk 3]
```diff
@@ -176,8 +178,17 @@ public Throwable getNotificationOperationFailure() {
...
 
         @Override
+        public Object getNotificationOperationDetails() {
+            return details;
+        }
```
Comment (etiennestuder) [inline subprojects/core/src/main/java/org/gradle/internal/operations/BuildOperationNotificationBridge.java]: Switch order to have `getNotificationOperationDetails ` after the id getter.

File: subprojects/core/src/main/java/org/gradle/internal/operations/BuildOperationNotificationBridge.java (hunk 1 above)
Comment (etiennestuder) [inline subprojects/core/src/main/java/org/gradle/internal/operations/BuildOperationNotificationBridge.java]: `details` after `id`

File: subprojects/core/src/main/java/org/gradle/internal/operations/BuildOperationNotificationBridge.java [hunk 4]
```diff
@@ -153,11 +153,13 @@ public String toString() {
         private final Object id;
         private final Object result;
         private final Throwable failure;
+        private final Object details;
```
Comment (etiennestuder) [inline subprojects/core/src/main/java/org/gradle/internal/operations/BuildOperationNotificationBridge.java]: Switch order to have `details` after the `id`.

File: subprojects/base-services/src/main/java/org/gradle/internal/operations/notify/BuildOperationFinishedNotification.java [hunk 5]
```diff
@@ -47,4 +47,9 @@
      */
     Throwable getNotificationOperationFailure();
 
+    /**
+     * A structured object providing details about the operation that was performed.
+     */
+    Object getNotificationOperationDetails();
+
```
Comment (etiennestuder) [inline subprojects/base-services/src/main/java/org/gradle/internal/operations/notify/BuildOperationFinishedNotification.java]: Switch order to have `getNotificationOperationDetails ` after the id getter.

File: subprojects/core/src/main/java/org/gradle/internal/operations/BuildOperationNotificationBridge.java [hunk 6]
```diff
@@ -153,11 +153,13 @@ public String toString() {
...
 
-        private Finished(Object id, Object result, Throwable failure) {
+        private Finished(Object id, Object result, Throwable failure, Object details) {
```
Comment (etiennestuder) [inline subprojects/core/src/main/java/org/gradle/internal/operations/BuildOperationNotificationBridge.java]: Switch order to have `details` after the `id`.

File: subprojects/core/src/test/groovy/org/gradle/internal/operations/BuildOperationNotificationBridgeTest.groovy [hunk 7]
```diff
@@ -152,6 +153,7 @@ class BuildOperationNotificationBridgeTest extends Specification {
             assert n.notificationOperationId == d3.id
             assert n.notificationOperationResult == null
             assert n.notificationOperationFailure == e1
+            assert n.notificationOperationDetails.is(d3.details)
```
Comment (etiennestuder) [inline subprojects/core/src/test/groovy/org/gradle/internal/operations/BuildOperationNotificationBridgeTest.groovy]: In all there blocks, switch order to have `details` after the `id`.

**Comment by alkemist** [comment]
@etiennestuder I've fixed those things locally. Will push in a bit.

**Pull request merged**